source/collections/Queue.ooc

```diff
@@ -12,17 +12,20 @@ Queue: abstract class <T> {
 	dequeue: abstract func -> (T, Bool)
 	peek: abstract func -> (T, Bool)
 	//TODO: This works much better and will do for now
-	dequeue: abstract func ~out (result: T*) -> Bool
+	dequeue: abstract func ~out (result: T*) -> Bool // Also crashing the compiler
```

Comment:
It is hard to understand why it crashed.

"Functional" means returning by value instead of returning by reference to target. What name do you suggest?